lib/drb/drb.rb

```diff
@@ -1527,7 +1553,13 @@ def insecure_method?(msg_id)
     def any_to_s(obj)
       obj.to_s + ":#{obj.class}"
     rescue
-      sprintf("#<%s:0x%lx>", obj.class, obj.__id__)
+      case obj
+      when Object
+        klass = obj.class
+      else
+        klass = Kernel.instance_method(:class).bind(obj).call
+      end
+      sprintf("#<%s:0x%dx>", klass, obj.__id__)
```

Comment:
`"#<%s:0x%dx>"` is still incorrect. `%d` stands for **d**ecimal (and the `x` is just a literal `x`) but it seems he**x**adecimal is the intended conversion here: `"#<%s:0x%x>"` or even `"#<%s:%#x>"`.

This method seems to based on pre-1.9 `rb_any_to_s()` (e.g. see the `l` modifier that C has for `long`) but it works differently even though it looks similar. However `rb_any_to_s()` is also the implementation of `Kernel#to_s`. Any reason not to just bind+call it like other standard libraries do?